### ISSUE TITLE
progress_callback arguments for *_async() functions have incorrect scope annoations.

### ISSUE THREAD

**Issue by jhenstridge** [canonical/snapd-glib#110]
This isn't a problem I'm blocked on or anything: just a bug I noticed when reading through the code.

All the async functions declare their progress callback as:

```
 * @progress_callback: (allow-none) (scope call): function to callback with progress.
```

This tells a language binding that the callback will not be called after the return of the function call, so it is safe to free any ffi closures at that point. That's clearly incorrect here, as we'd expect to see the progress callback called multiple times after the function returns, right up until the async ready callback fires.

Neither of the [other two scope types](https://gi.readthedocs.io/en/latest/annotations/giannotations.html#type-signature) look to be appropriate:

* `scope async` callbacks may only be called once, with the language binding being free to clean up after that call.
* `scope notified` is the best fit (allows multiple calls after return from the function), but we don't provide a GDestroyNotify for the progress callback that it requires.

So it's not clear this can be solved purely by updating the annotations.

The synchronous functions look like they should be okay with `scope call` though: the progress callback will never be called again after those functions return.

**Comment by sergio-costas** [comment]
I presume that this would require a change in the API to include the GDestroyNotify parameter, isn't it?

**Comment by robert-ancell** [comment]
Yeah, for bindings to correctly handle this we'd have add an additional parameter, possibly by making new `_full()` functions. I don't know of any bindings in use currently though.

**Comment by 3v1n0** [comment]
This has been "fixed" via 27915d17aff4ffc64cb2c1b83b8e9d68a0fa5f21 and 419dd89c04f9a52f1618df9bb873e711b7b74bf9 while indeed a proper fix would require an API change.

We could preserve the ABI though, if we use versioned-symbols properly.
